code/modules/mapping/space_management/space_transition.dm

```diff
@@ -162,4 +147,61 @@
 				var/turf/place = locate(S.destination_x, S.destination_y, zdestination)
 				S.AddElement(/datum/element/mirage_border, place, mirage_dir, range_cached)
 
+/// Construct linkages randomly to get confusing space transitions
+/// We do this by constructing an absurdly large grid, and placing the levels randomly inside
+/// Please update this comment when you figure out why this is
```

Comment:
can you elaborate on what part of this bugs you, I at once point understood this code so I might be able to help

Reply:
I actually kind of understand it now too. I should swap this comment out with something more useful